### ISSUE TITLE
undefined reference to `Caml_state' with OCaml 5.2.1

### ISSUE THREAD

**Issue by Lupus** [zshipko/ocaml-rs#156]
I have this weird linking error in one of my projects:

```
+ gcc -O2 -fno-strict-aliasing -fwrapv -pthread  -pthread  -Wl,-E  -o 'test/test.exe'  '-Lrust-staticlib' '-L/home/kolkhovskiy/git/rust/ocaml-rs-smartptr/_opam/lib/ocaml'  '/tmp/build_9aeded_dune/camlstartup207a33.o' '/home/kolkhovskiy/git/rust/ocaml-rs-smartptr/_opam/lib/ocaml/std_exit.o' 'test/.test.eobjs/native/dune__exe__Test.o' 'test/.test.eobjs/native/dune__exe__Stubs.o' 'test/.test.eobjs/native/dune__exe.o' 'lib/ocaml_rs_smartptr.a' 'rust-staticlib/ocaml_rs_smartptr_stubs.a' '/home/kolkhovskiy/git/rust/ocaml-rs-smartptr/_opam/lib/ocaml/stdlib.a' '-lrust_staticlib_ocaml_rs_smartptr' '-lpthread' '-lc' '-lm' '/home/kolkhovskiy/git/rust/ocaml-rs-smartptr/_opam/lib/ocaml/libasmrun.a'   -lm -ldl  -lpthread
/usr/bin/ld: rust-staticlib/librust_staticlib_ocaml_rs_smartptr.a(ocaml_sys-86d3e9f8a7cfa4c4.ocaml_sys.11c84160b7b507b6-cgu.0.rcgu.o): in function `ocaml_sys::state::local_roots':
/home/kolkhovskiy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ocaml-sys-0.24.0/src/state.rs:150: undefined reference to `Caml_state'
/usr/bin/ld: rust-staticlib/librust_staticlib_ocaml_rs_smartptr.a(ocaml_sys-86d3e9f8a7cfa4c4.ocaml_sys.11c84160b7b507b6-cgu.0.rcgu.o): in function `ocaml_sys::state::set_local_roots':
/home/kolkhovskiy/.cargo/registry/src/index.crates.io-6f17d22bba15001f/ocaml-sys-0.24.0/src/state.rs:155: undefined reference to `Caml_state'
collect2: error: ld returned 1 exit status
File "caml_startup", line 1:
Error: Error during linking (exit code 1)
```

I see that the symbol is indeed undefined in static library built for Rust stubs:

```
kolkhovskiy@kolkhovskiy-dev:~/git/rust/ocaml-rs-smartptr$ nm _build/default/rust-staticlib/librust_staticlib_ocaml_rs_smartptr.a | grep Caml_state
                 U Caml_state
```

But other project in different switch, but on the same system and same OCaml 5.2.1 compiles just fine:

```
+ gcc -O2 -fno-strict-aliasing -fwrapv -pthread  -pthread  -Wl,-E  -o 'tests/ocamlgen_test.exe'  '-L/home/kolkhovskiy/git/rust/ocaml-gen/_opam/lib/ocaml'  '/tmp/build_faa764_dune/camlstartup6ca22d.o' '/home/kolkhovskiy/git/rust/ocaml-gen/_opam/lib/ocaml/std_exit.o' 'tests/.ocamlgen_test.eobjs/native/dune__exe__Ocamlgen_test.o' 'tests/.ocamlgen_test.eobjs/native/dune__exe.o' '/home/kolkhovskiy/git/rust/ocaml-gen/_opam/lib/ocaml/stdlib.a' 'tests/libocamlgen_test_stubs.a' '/home/kolkhovskiy/git/rust/ocaml-gen/_opam/lib/ocaml/libasmrun.a'   -lm -ldl  -lpthread
```

Same `Caml_state` is undefined in Rust static library with stubs:

```
kolkhovskiy@kolkhovskiy-dev:~/git/rust/ocaml-gen$ nm _build/default/tests/libocamlgen_test_stubs.a | grep Caml_state
                 U Caml_state
```

I'm quite confused on what to do about this. Also it seems that `Caml_state` is not used by `ocaml-rs` per se... I wasn't able to trace down which object file or library should define that Caml_state, fiddling with dune libs didn't help - I tried depending on `domainslib` and `threads` but it still fails with linker error :(

**Comment by Lupus** [comment]
It seems that changes in https://github.com/zshipko/ocaml-rs/pull/157 are sufficient for this to work. @zshipko please take a look if that makes any sense!

**Comment by zshipko** [comment]
I've just published a new release after merging #157, this should be fixed now. Thanks again @Lupus!
